code/core_save.c

```diff
@@ -14,12 +14,16 @@
 // keep a static file handle to remove the need to do constant seeking.
 FILE* save_fd[ABSOLUTEMAXSNAPS] = { 0 };
 
+#define MAX_BUF_SIZE (2*MAX_STRING_LEN+40)
+#define MAX_OUTFILE_SIZE (MAX_STRING_LEN+40)
 
 void save_galaxies(int filenr, int tree)
 {
   char buf[MAX_STRING_LEN];
   int i, n;
-  struct GALAXY_OUTPUT galaxy_output = {0};
+  static const struct GALAXY_OUTPUT galaxy_output_null = {0};
+  struct GALAXY_OUTPUT galaxy_output = galaxy_output_null;
```

Comment:
I am curious - what's the advantage of splitting up this way?

Reply:
Strictly speaking, you're welcome to drop that, depending on how interested you are in future integration with our [elaphrocentre](https://codeberg.org/boud/elaphrocentre) galaxy formation pipeline, which we just posted to [ArXiv:2010.03742](https://arXiv.org/abs/2010.03742) a few days ago. If you look at line 1042 of Marius' @mpeper patch

https://codeberg.org/boud/elaphrocentre/src/branch/N128cubed/reproduce/software/patches/marius_sage_hack.patch#L1042

which is applied before the other two patches at lines 1610--1612 of the current commit of

https://codeberg.org/boud/elaphrocentre/src/branch/N128cubed/reproduce/software/make/high-level.mk#L1610

then you'll see why we did that. For a given "final" halo at the current epoch, we add up the mass infall histories of all the predecessor haloes. Since this is a loop through "final" haloes, it's safer to zero the galaxy_output structure between iterations.

You're welcome to come over to codeberg and propose improvements to our sage patches, especially for more modularity. Or you can try to convince @mpeper to modularise and tidy his patch enough to propose a pull request here ;).